samcli/commands/build/build_context.py

```diff
@@ -344,29 +348,53 @@ def _handle_build_post_processing(self, builder: ApplicationBuilder, build_resul
 
             move_template(stack.location, output_template_path, modified_template)
 
-    @staticmethod
-    def gen_success_msg(artifacts_dir: str, output_template_path: str, is_default_build_dir: bool) -> str:
+    def _gen_success_msg(self, artifacts_dir: str, output_template_path: str, is_default_build_dir: bool) -> str:
+        """
+        Generates a success message containing some suggested commands to run
 
-        invoke_cmd = "sam local invoke"
-        if not is_default_build_dir:
-            invoke_cmd += " -t {}".format(output_template_path)
+        Parameters
+        ----------
+        artifacts_dir: str
+            A string path representing the folder of built artifacts
+        output_template_path: str
+            A string path representing the final template file
+        is_default_build_dir: bool
+            True if the build folder is the folder defined by SAM CLI
+
+        Returns
+        -------
+        str
+            A formatted success message string
+        """
+
+        validate_suggestion = "Validate SAM template: sam validate"
+        invoke_suggestion = "Invoke Function: sam local invoke"
+        sync_suggestion = "Test Function in the Cloud: sam sync --stack-name {{stack-name}} --watch"
+        deploy_suggestion = "Deploy: sam deploy --guided"
+        start_lambda_suggestion = "Emulate local Lambda functions: sam local start-lambda"
 
-        deploy_cmd = "sam deploy --guided"
         if not is_default_build_dir:
-            deploy_cmd += " --template-file {}".format(output_template_path)
+            invoke_suggestion += " -t {}".format(output_template_path)
+            deploy_suggestion += " --template-file {}".format(output_template_path)
+
+        commands = [validate_suggestion, invoke_suggestion, sync_suggestion, deploy_suggestion]
+
+        # check if we have used a hook package before building
+        if self._hook_package_id:
+            hook_package_flag = f" --hook-package-id {self._hook_package_id}"
+            start_lambda_suggestion += hook_package_flag
```

Comment:
nit: it works without the hook package flag, but should we allow add it for the local invoke command?

Reply:
it will work, but let's here at least printed with the `--hook-package-id ` ... as in case if the project does not need to be built, the customers have to provide it.

Reply:
Nice catch, will add hook package flag to invoke